includes/addon-moderation.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * Add a BP Docs-specific "pending" status to use in a moderation workflow.
+ *
+ * @since 2.1.0
+ */
+class BP_Docs_Moderation {
+
+	public $pending_status = 'bp_docs_pending';
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 2.1.0
+	 */
+	public function __construct() {
+	}
+
+	/**
+	 * Hook the Folders functionality into Docs.
+	 *
+	 * @since 2.1.0
+	 */
+	public function add_hooks() {
+		// Register a custom status that's a lot like the built-in "pending" status.
+		add_action( 'bp_docs_init', array( $this, 'register_docs_pending_status' ), 9 );
+
+		add_filter( 'display_post_states', array( $this, 'add_moderated_label' ), 10, 2 );
+	}
+
+	/**
+	 * Register custom status for pending BP Docs.
+	 *
+	 * @since 2.1.0
+	 */
+	public function register_docs_pending_status() {
+		$args = array(
+			'label'                     => _x( 'Awaiting Moderation', 'Status General Name', 'buddypress-docs' ),
+			'label_count'               => _n_noop( 'Awaiting Moderation (%s)',  'Pending (%s)', 'buddypress-docs' ),
```

Comment:
`_n_noop()` is for registering singular + plural - why are the singular and plural different?

Reply:
It seems that `register_post_status()` or rather the global `$wp_post_statuses` expects the label_count parameter to be an `_n_noop()`. They shouldn't be different, though, that's just me being not very smart, ha ha.

https://developer.wordpress.org/reference/functions/register_post_status/